caps/capabilities_test.go

```diff
@@ -35,10 +35,13 @@ var _ = Suite(&CapabilitySuite{})
 
 func (s *CapabilitySuite) TestString(c *C) {
 	cap := &Capability{
-		Name:     "test-name",
+		ID: CapabilityID{
```

Comment:
They keying by (snap, capability) at this level of the API feels slightly curious. It's optimizing for a very particular lookup, which will become very fast but that doesn't seem to match many of the interactions we need to put in place soon (caps for a snap, providers, consumers, etc).

Besides that, there's also another issue which is worth some thought: a particular capability is not actually a property of a given snap name, but of a particular revision of that snap name. And we also have the two sides: a provider, and a consumer, and we have the linkage between the two. 

Finally, we should think carefully about introducing the concept of a capability id when we already have a capability name which is the identifier we want people to think about. We already have type and name.. having type, name, and id wouldn't help conversations about the topic.

I'll think a bit through this as well. Let's please catch up online.